ledger/accountdb.go

```diff
@@ -1989,6 +1995,41 @@ func (qs *accountsDbQueries) lookupKeyValue(key string) (pv persistedValue, err
 	return
 }
 
+func (qs *accountsDbQueries) lookupKeysByPrefix(prefix string, maxKeyNum uint64, results map[string]bool, resultCount uint64) (round basics.Round, err error) {
+	err = db.Retry(func() (_err error) {
+		rows, _err := qs.lookupKeysByPrefixStmt.Query(len(prefix), []byte(prefix))
+		if _err != nil {
+			if _err == sql.ErrNoRows {
+				// we just found nothing, not an error I guess?
+				_err = nil
+				return
```

Comment:
I think this just as bad as in the other lookup functions. Because we are LEFT JOIN with the acctrounds table, we should _always_ get a row. So think this should have the same error behavior:
```
			// this should never happen; it indicates that we don't have a current round in the acctrounds table.
			if err == sql.ErrNoRows {
				// Return the zero value of data
				err = fmt.Errorf("unable to query key for %v : %w", prefix, err)
			}
			return err
```

Reply:
On the other hand, I'm not sure that `Query()` _can_ return sql.ErrNoRows (as opposed to `QueryRow`, which we use in the others).  So maybe we should not check against `sql.ErrNoRows` at all.

Reply:
did a quick googling: https://stackoverflow.com/questions/60123848/does-db-query-return-errnorows, seems the error handling is unnecessary?

Reply:
tldr it means error of `sql.ErrNoRows` derives from `Scan` not `Query`, so in other cases we do `QueryRow(...).Scan(...)`, then yes indeed we do need to have error handling.

but in this case, we just do `if _err != nil { return }` and should be fine.

Reply:
you want `return _err`, surely?